atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -155,6 +155,15 @@
 @Value.Immutable
 @StagedBuilderStyle
 public abstract class TransactionManagers {
+    static {
+        String javaVersion = System.getProperty("java.version");
+        Preconditions.checkState(
+                !javaVersion.startsWith("21"),
```

Comment:
good spot: `1.8.0_210` is unlikely, but would be fun...

Reply:
Perhaps `Runtime.version().feature() == 21`?

Reply:
Note that it will also break libraries which attempt to test against jdk21

Reply:
Only if those libraries themselves use Atlas (i.e., our internal migration and queue libraries). Do we explicitly not want to break those? The migration library for one is already broken under 21 (and that's how we got here)

Reply:
Yep, I think it's reasonable, I only wanted to document the edge case :-)